tests/testTransforms.py

```diff
@@ -0,0 +1,468 @@
+
+def makeRandomScaledPolynomialTransform(order):
+    return ScaledPolynomialTransform(
+        makeRandomPolynomialTransform(order),
+        makeRandomAffineTransform(),
+        makeRandomAffineTransform()
+    )
+
+def makeRandomSipForwardTransform(order):
+    return SipForwardTransform(
+        lsst.afw.geom.Point2D(*numpy.random.randn(2)),
+        lsst.afw.geom.LinearTransform(numpy.random.randn(2,2)),
+        makeRandomPolynomialTransform(order, sip=True)
+    )
+
+def makeRandomSipReverseTransform(order):
+    origin = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+    cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+    poly = makeRandomPolynomialTransform(order, sip=False)
+    return SipReverseTransform(origin, cd, poly)
+
+
+class TransformTestMixin(object):
+
+    def checkTransformEquivalence(self, a, b, rtol=1E-8):
+        for i in xrange(10):
+            point = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+            aOut = a(point)
+            bOut = b(point)
+            self.assertClose(numpy.array(aOut), numpy.array(bOut), rtol=rtol)
+
+    def testLinearize(self):
+        """Test that the AffineTransform returned by linearize() is equivalent
+        to the transform at the expansion point, and matches finite differences.
+        """
+        transform = self.makeRandom();
+        point = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+        affine = transform.linearize(point)
+        self.assertClose(numpy.array(transform(point)), numpy.array(affine(point)), rtol=1E-14)
+        delta = 1E-4
+        deltaX = lsst.afw.geom.Extent2D(delta, 0.0)
+        deltaY = lsst.afw.geom.Extent2D(0.0, delta)
+        dtdx = (transform(point + deltaX) - transform(point - deltaX)) / (2*delta)
+        dtdy = (transform(point + deltaY) - transform(point - deltaY)) / (2*delta)
+        self.assertClose(affine[affine.XX], dtdx.getX(), rtol=1E-6)
+        self.assertClose(affine[affine.YX], dtdx.getY(), rtol=1E-6)
+        self.assertClose(affine[affine.XY], dtdy.getX(), rtol=1E-6)
+        self.assertClose(affine[affine.YY], dtdy.getY(), rtol=1E-6)
+
+
+class PolynomialTransformTestCase(lsst.utils.tests.TestCase, TransformTestMixin):
+
+    def setUp(self):
+        numpy.random.seed(50)
+
+    def makeRandom(self):
+        return makeRandomPolynomialTransform(4)
+
+    def testArrayConstructor(self):
+        """Test that construction with coefficient arrays yields an object with
+        copies of those arrays, and that all dimensions must be the same.
+        """
+        order = 3
+        xc = makeRandomCoefficientMatrix(order + 1)
+        yc = makeRandomCoefficientMatrix(order + 1)
+        p = PolynomialTransform(xc, yc)
+        self.assertEqual(p.getOrder(), order)
+        self.assertClose(p.getXCoeffs(), xc, atol=0, rtol=0)
+        self.assertClose(p.getYCoeffs(), yc, atol=0, rtol=0)
+        # Test that the coefficients are not a view.
+        xc[0, 0] = 50.0
```

Comment:
It is not self-evident that the existing value cannot be 50. Please consider incrementing the existing value (after saving the initial value to use in the comparison test).